pkg/provisioner/ironic/ironic.go

```diff
@@ -1116,6 +1119,21 @@ func (p *ironicProvisioner) Prepare(data provisioner.PrepareData, unprepared boo
 			result, err = p.setMaintenanceFlag(ironicNode, false)
 			return
 		}
+		if ironicNode.TargetProvisionState == string(nodes.Available) {
+			result, err = p.changeNodeProvisionState(
+				ironicNode,
+				nodes.ProvisionStateOpts{Target: nodes.TargetManage},
+			)
+			if err != nil {
+				return
```

Comment:
@hs0210 Maybe we should add log entry here?

Reply:
Thank you for your review, I will add the corresponding log output later.

Reply:
Sorry for not updating for a long time, I did some change with the logic, please review again.